File: saransh_kunal-LINUX/Assignment/Enterprise/EFMvc/DemoApp1/Tourism/Models/SiteModel.cs
using Shopping.Data;
// using Tourism.Data;

namespace Tourism.Models;

public class SiteModel
{
    public IEnumerable<Candidate> GetCandidate()
    {
        using var site = new EmployeeDbContext();
        var selection = from t in site.departments
                        // where t.Id.Length > 1
                        select new Candidate
                        {
                              deptId = t.Id,
                            dname = t.dname,
                           loc = t.loc
                        };
        return selection.ToList();
    }

    public void AcceptVisit(decimal candidateId)
    {
        using var site = new EmployeeDbContext();
        var dept = site.departments.Find(candidateId);
        if(dept is null)
        {
            dept = new Department { Id = candidateId };
            site.departments.Add(dept);
        }
        dept.det.Add(new Employee());
        site.SaveChanges();
    }
}

File: saransh_kunal-LINUX/Assignment/java/Advanced/assignment_8/src/main/java/app/company/models/SiteModel.java
package app.company.models;

import app.company.data.DepartmentEntity;
import app.company.data.EmployeeEntity;
import jakarta.persistence.EntityManagerFactory;
import jakarta.persistence.Persistence;

public class SiteModel {
    
    private static EntityManagerFactory emf = Persistence.createEntityManagerFactory("app.data");

    public Iterable<Candidate> getEmployee() {
        try(var em = emf.createEntityManager()){
            var query = em.createQuery("SELECT t FROM DepartmentEntity t", DepartmentEntity.class);
            return query.getResultStream()
                .map(Candidate::fromTraveler)
                .toList();
        }
    }

    public boolean acceptVisit(int deptno, String dname, String loc) {
      
        try(var em = emf.createEntityManager()){
            var department = em.find(DepartmentEntity.class, deptno);
            if(department == null){
                department = new DepartmentEntity();
                department.setDeptno(deptno);
            }
            department.setDNAME(dname);
            department.setLOC(loc);
            var emp = new EmployeeEntity();
            emp.setDeptno(department);
            department.getEmployee().add(emp);
            var tx = em.getTransaction();
            tx.begin();
            em.persist(department);
            tx.commit();
            return true;
        }
    }

}
